### PR TITLE
Add .NET 7 SDK extension

### PR DESCRIPTION
**Pull request by Dryadxon**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [ ] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [ ] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+  "only-arches": ["x86_64", "aarch64"],
```
Comment (hfiguiere) [inline flathub.json:2]:
```suggestion
```

there is no other arch

File: flathub.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+  "skip-icons-check": true,
+  "disable-external-data-checker": true
```
Comment (hfiguiere) [inline flathub.json:4]: why not using the built-in one?

File: org.freedesktop.Sdk.Extension.dotnet7.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      <description>
+Included runtimes
+.NET Runtime 7.0.0
+ASP.NET Core Runtime 7.0.0
+.NET Desktop Runtime 7.0.0
+
+Language support
+C# 11.0
+F# 7.0
+Visual Basic 16.9
```
Comment (hfiguiere) [inline org.freedesktop.Sdk.Extension.dotnet7.appdata.xml:26]: you might want to use HTML list here.

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by hfiguiere** [comment]
bot, build org.freedesktop.Sdk.Extension.dotnet7

File: flathub.json (hunk 1 above)
Comment (Dryadxon) [reply flathub.json:2]: I simply left the file as in the .NET 6 SDK repository and changed the minimum necessary for the new major version. But I also agree with removing it since, as you said, there are no other architectures to distribute.

File: flathub.json (hunk 2 above)
Comment (Dryadxon) [reply flathub.json:4]: This is done to be "future-proof" regarding a future release of the runtime `23.08`, matching what has been done for the .NET 6 SDK with the `21.08` and `22.08` runtimes. For reference, [here is the PR] (https://github.com/flathub/org.freedesktop.Sdk.Extension.dotnet6/pull/32) which describes why this workaround is used to keep both branches updated
Of course, it isn't a problem currently, since it isn't out yet, I just wanted to set it once and forget it to be completely honest.

File: org.freedesktop.Sdk.Extension.dotnet7.appdata.xml (hunk 3 above)
Comment (Dryadxon) [inline org.freedesktop.Sdk.Extension.dotnet7.appdata.xml:26]: Something like that?

```html
Included runtimes
<ul>
  <li>.NET Runtime 7.0.0</li>
  <li>ASP.NET Core Runtime 7.0.0</li>
  <li>.NET Desktop Runtime 7.0.0</li>
</ul>

Language support
<ul>
  <li>C# 11.0</li>
  <li>F# 7.0</li>
  <li>Visual Basic 16.9</li>
</ul>
```

Comment (Dryadxon) [reply org.freedesktop.Sdk.Extension.dotnet7.appdata.xml:26]:
```html
Included runtimes
<ul>
  <li>.NET Runtime 7.0.0</li>
  <li>ASP.NET Core Runtime 7.0.0</li>
  <li>.NET Desktop Runtime 7.0.0</li>
</ul>

Language support
<ul>
  <li>C# 11.0</li>
  <li>F# 7.0</li>
  <li>Visual Basic 16.9</li>
</ul>
```
Do you mean something like that?

**Review by Dryadxon — COMMENTED** [review]
I replaced the `<description>` content to the supported HTML as suggested.

File: org.freedesktop.Sdk.Extension.dotnet7.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+          version-pattern: ^([\d\.a-z-]+)$
+          url-template: https://dotnetcli.blob.core.windows.net/dotnet/Sdk/$version/dotnet-sdk-$version-linux-arm64.tar.gz
+  - name: scripts
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /usr/lib/sdk/dotnet7/share/appdata
+      - cp install.sh install-sdk.sh /usr/lib/sdk/dotnet7/bin
+      - cp enable.sh /usr/lib/sdk/dotnet7
+      - cp org.freedesktop.Sdk.Extension.dotnet7.appdata.xml ${FLATPAK_DEST}/share/appdata
+      - appstream-compose --basename=org.freedesktop.Sdk.Extension.dotnet7 --prefix=${FLATPAK_DEST}
```
Comment (JakobDev) [inline org.freedesktop.Sdk.Extension.dotnet7.yaml:42]: Why did you need this? flatpak-builder should do this automatic.

Comment (barthalion) [reply org.freedesktop.Sdk.Extension.dotnet7.yaml:42]: Because it doesn't for extensions.

**Comment by barthalion** [comment]
bot, build org.freedesktop.Sdk.Extension.dotnet7

File: org.freedesktop.Sdk.Extension.dotnet7.yaml (hunk 4 above)
Comment (JakobDev) [reply org.freedesktop.Sdk.Extension.dotnet7.yaml:42]: Maybe it is better to use appstreamcli here instead of the old appstream-compose

Comment (hfiguiere) [reply org.freedesktop.Sdk.Extension.dotnet7.yaml:42]: we use the same as elsewhere. Every other extension do that.
